builder/virtualbox/common/hw_config.go

```diff
@@ -0,0 +1,46 @@
+package common
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/packer/template/interpolate"
+)
+
+type HWConfig struct {
+
+	// cpu information
+	CpuCount   int `mapstructure:"cpus"`
+	MemorySize int `mapstructure:"memory"`
+
+	// device presence
+	Sound string `mapstructure:"sound"`
+	USB   bool   `mapstructure:"usb"`
+}
+
+func (c *HWConfig) Prepare(ctx *interpolate.Context) []error {
+	var errs []error
+
+	// Hardware and cpu options
+	if c.CpuCount < 0 {
+		errs = append(errs, fmt.Errorf("An invalid number of cpus was specified (cpus < 0): %d", c.CpuCount))
+		c.CpuCount = 0
```

Comment:
This set is probably not necessary since we error

```suggestion
```

Reply:
Yea, I assigned those since a user can still call `.Prepare()` and not check the error. So immediately afterwards it'll still fix the `CpuCount` to 1 if the user wants to use the config despite it erroring out. Nonetheless, though, I added your changes.

Reply:
Oh right !! didn't see that one !